source/functions.txt

```diff
@@ -213,12 +213,322 @@ UI or by importing the function configuration and source code with
+
+            Once you've written the function code, click :guilabel:`Save` from
+            either the :guilabel:`Function Editor` or :guilabel:`Settings` tab.
+
+            After you save the function, you can begin using it immediately.
+
    
    .. tab::
       :tabid: cli
       
-      .. include:: /includes/steps/define-a-function-import-export.rst
+      .. procedure::
+
+         .. step:: Pull Your App's Latest Configuration Files
+
+            .. code-block:: shell
+
+               realm-cli pull --remote=<App ID>
+
+
+         .. step:: Write the Function Source Code
+
+            {+service-short+} functions run standard ES6+ JavaScript functions that you export from
+            individual files. Create a ``.js`` file with the same name as the function in
+            the ``functions`` directory or one of its subdirectories.
+
+            .. code-block:: bash
+
+               touch functions/myFunction.js
+
+            .. tip::
+
+               You can define functions inside of nested folders in the
+               ``functions`` directory. Use slashes in a function name to indicate
+               its directory path.
+
+            Once you've created the function's ``.js`` file, write the function source code, e.g.:
+
+            .. code-block:: javascript
+
+               exports = async function hello(...args) {
+                 // Write your function logic here! You can...
+                 // Import dependencies
+                 const assert = require("assert")
+                 assert(typeof args[0] === "string")
+                 // Use ES6+ syntax
+                 const sayHello = (name = "world") => {
+                   console.log(`Hello, ${name}.`)
+                 }
+                 // Return values back to clients or other functions
+                 return sayHello(args[0])
+               }
+
+            .. note::
+
+               You can use most modern (ES6+) JavaScript features in functions, including
+               async/await, destructuring, and template literals. To see if {+service-short+} supports
+               a specific feature, see :ref:`JavaScript Support <js-support>`.
+
+
+         .. step:: Configure the Function
+
+            In the ``functions`` directory of your application, open the ``config.json``
+            file and add a :ref:`configuration object <appconfig-functions>` for your new
+            function to the array. The object must have the following form:
+
+            .. code-block:: json
+
+               {
+                 "name": "<Function Name>",
+                 "private": <Boolean>,
+                 "can_evaluate": { <JSON Expression> },
+                 "disable_arg_logs": <Boolean>,
+                 "run_as_system": <Boolean>,
+                 "run_as_user_id": "<{+service-short+} User ID>",
+                 "run_as_user_id_script_source": "<Function Source Code>"
+               }
+
+         .. step:: Set User Authentication
```

Comment:
I separated them out because I don't think headings are compatible with the procedure/step directives -- I've never seen rST display them that way. Will make this a series of sub-steps instead.